packages/react/src/xcss-prop/index.ts

```diff
@@ -145,23 +171,35 @@ export type XCSSAllPseudos = CSSPseudos;
  * <Component xcss={styles.text} />
  * ```
  *
- * To concatenate and conditonally apply styles use the {@link cssMap} {@link cx} functions.
+ * To concatenate and conditionally apply styles use the {@link cssMap} {@link cx} functions.
  */
 export type XCSSProp<
-  TAllowedProperties extends keyof StrictCSSProperties,
-  TAllowedPseudos extends CSSPseudos,
+  TAllowedProperties extends keyof StrictCSSProperties | StrictCSSProperties, // extend StrictCSSProperties with CSS vars.
+  TAllowedPseudos extends TAllowedProperties extends object ? never : CSSPseudos,
   TRequiredProperties extends {
-    requiredProperties: TAllowedProperties;
+    requiredProperties: TAllowedProperties extends object ? never : TAllowedProperties;
   } = never
 > = Internal$XCSSProp<
-  TAllowedProperties,
-  TAllowedPseudos,
+  TAllowedProperties extends object ? FilterPseudos<keyof TAllowedProperties> : TAllowedProperties,
```

Comment:
On second thought, wouldn't this work?  We already use this syntax and have the keys to pick this side of it 🤦 my bad, anyways, still helps to filter/extract here.
```suggestion
  TAllowedProperties extends object
    ? Extract<keyof TAllowedProperties, keyof StrictCSSProperties>
    : TAllowedProperties,
```